api/allocations.go

```diff
@@ -127,6 +127,16 @@ func (a *Allocations) Restart(alloc *Allocation, taskName string, q *QueryOption
 	return err
 }
 
+func (a *Allocations) RestartAllTasks(alloc *Allocation, q *QueryOptions) error {
```

Comment:
I created a new method here to avoid an `api` package breaking change in the `Restart` method above.